### PR TITLE
simplex-chat-desktop: init at 5.3.1

### PR DESCRIPTION
**Pull request by yuuyins**
## Description of changes

add simplex-chat-desktop AppImage https://github.com/simplex-chat/simplex-chat/releases/tag/v5.3.1

simpex-chat-desktop.nix instead of default.nix because there are other clients like the CLI which someone might add in the future in that same directory...

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by emilylange** [comment]
Could this be built from source instead? :)

**Comment by yuuyins** [comment]
@emilylange TLDR: I am all for building from source, but I am afraid considering the risks it would likely take months of my effort for this, so I do not see myself trying for now, maybe if I have spare time in the future as this is an app I will be using very much.

Longer: It is just that would probably involve some complex build-process involving gradle for Kotlin, besides considering this mono-repo with sharing code for multi-plataform https://github.com/simplex-chat/simplex-chat/tree/stable/apps/multiplatform... and from experience from last years trying, JVM builders are little to no supported on Nix/Nixpkgs, plus a ton of issues with JVM-based software. So in my experience, it is not worth the risk of such effort trying to package it to end up in failure (which happened with me multiple times).

File: pkgs/applications/networking/instant-messengers/simplex-chat/simplex-chat-desktop.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      cp -r ${appimageContents}/usr/share/icons $out/share
+    '';
+
+  meta = with lib; {
+    description = "Desktop application for SimpleX Chat";
+    homepage = "https://simplex.chat";
+    changelog = "https://github.com/simplex-chat/simplex-chat/releases/tag/v${version}";
+    license = licenses.agpl3Only;
+    maintainers = with maintainers; [ yuu ];
+    platforms = with platforms; [ "x86_64-linux" ];
```
Comment (emilylange) [inline pkgs/applications/networking/instant-messengers/simplex-chat/simplex-chat-desktop.nix:44]: `lib.platforms` isn't needed here

```suggestion
    platforms = [ "x86_64-linux" ];
```

File: pkgs/applications/networking/instant-messengers/simplex-chat/simplex-chat-desktop.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{ lib
+  , appimageTools
+  , fetchurl
```
Comment (emilylange) [inline pkgs/applications/networking/instant-messengers/simplex-chat/simplex-chat-desktop.nix:3]: Nitpick

```suggestion
, appimageTools
, fetchurl
```

**Review by emilylange — CHANGES_REQUESTED** [review]
Please also move this package from `all-packages.nix` into the new `by-name` scheme.
See [`pkgs/by-name/README.md`](https://github.com/NixOS/nixpkgs/blob/0fbc32585c7617c1fb7b7acbb1db9f13d54c3d04/pkgs/by-name/README.md)

Thanks!

File: pkgs/applications/networking/instant-messengers/simplex-chat/simplex-chat-desktop.nix (hunk 1 above)
Comment (yuuyins) [reply pkgs/applications/networking/instant-messengers/simplex-chat/simplex-chat-desktop.nix]: done that but what is going on with the `simplex-chat-desktop, simplex-chat-desktop.passthru.tests on aarch64-darwin` that didn't skip? https://logs.ofborg.org/?attempt_id=e105b7db-ecda-46fc-b43a-3167e2899e77&key=nixos%2Fnixpkgs.256615

**Comment by yuuyins** [comment]
@ofborg eval

**Comment by yuuyins** [comment]
@ofborg build

File: pkgs/top-level/all-packages.nix
Comment (emilylange) [inline pkgs/top-level/all-packages.nix:1]: This entry can be removed now, as `by-name` imports packages automatically based on its directory structure :)

**Review by emilylange — CHANGES_REQUESTED** [review]

**Review by emilylange — APPROVED** [review]

**Pull request merged**